tests/data_context/store/test_configuration_store.py

```diff
@@ -230,3 +250,102 @@ def test_v3_configuration_store(tmp_path_factory):
         )
         == 0
     )
+
+
+@pytest.mark.unit
+def test_overwrite_existing_property_and_setter() -> None:
+    store = SampleConfigurationStore(store_name="my_configuration_store")
+
+    assert store.overwrite_existing is False
+    store.overwrite_existing = True
+    assert store.overwrite_existing is True
+
+
+@pytest.mark.unit
+def test_config_property() -> None:
```

Comment:
nit: maybe call this `test_config_property_and_defaults()` or a seprate test for the defaults?